core/src/test/scala/org/apache/spark/sql/delta/MergeIntoScalaSuite.scala

```diff
@@ -53,6 +55,30 @@ class MergeIntoScalaSuite extends MergeIntoSuiteBase  with DeltaSQLCommandTest
     }
   }
 
+  test("basic scala explain") {
+    withTable("source") {
+      append(Seq((1, 10), (2, 20)).toDF("key1", "value1"), Nil)  // target
+      val source = Seq((1, 100), (3, 30)).toDF("key2", "value2")  // source
+
+      val builder = io.delta.tables.DeltaTable.forPath(spark, tempPath)
+        .merge(source, "key1 = key2")
+        .whenMatched().updateExpr(Map("key1" -> "key2", "value1" -> "value2"))
+        .whenNotMatched().insertExpr(Map("key1" -> "key2", "value1" -> "value2"))
+
+      val out = new ByteArrayOutputStream()
+      Console.withOut(out) {
+        builder.explain()
+      }
+
+      val explainOutput = out.toString("utf-8")
+      assert(explainOutput != null)
+      assert(explainOutput.contains("Execute MergeIntoCommand"))
+      assert(explainOutput.contains("+- MergeIntoCommand Project"))
+      assert(explainOutput.contains("Insert [actions:"))
```

Comment:
testing the exact formatting with `[actions` might be brittle. so maybe just test for "Insert". 

Also it would be good to rather see whether the all the columns names are being printed. So you can use different expressions with unique column names in each matched/unmatched clause to make sure each of the clause details are being printed.

Reply:
updated this to add regex for the delete, insert and update clauses and individual assertions on the specific fields. LMK if the changes are appropriate.